tools/scripts/generate.py

```diff
@@ -104,24 +104,91 @@ def build_yml(bug_id: str) -> None:
         logger.error(f"A file with the same name already exists: `{path}`, please edit it instead.")
         return
 
-    with open(path, 'w') as f:
-        yaml.dump(data, f, sort_keys=False, default_flow_style=False)
+    output_path = path if write else None
+    write_yml(data, output_path)
+    if write:
         logger.info(f"{filename}.yml file was created for bug {bug_id} ({title})")
+    return output_path
+
+
+def add_breakage(path: str, write: bool) -> Optional[str]:
+    with open(path) as f:
+        data = yaml.safe_load(f)
+    if "references" not in data:
+        data["references"] = {}
+
+    issues = data["references"].get("breakage", [])
+
+    reports = []
+    for item in issues:
+        if isinstance(item, str):
+            report = generate_breakage_report(item)
+            reports.append(report)
+        else:
+            reports.append(report)
```

Comment:
```suggestion
            reports.append(item)
```
Should this be `item` instead of `report`?